src/Orchard.Cms.Web/Modules/Orchard.OpenId/Views/Admin/Create.cshtml

```diff
@@ -16,61 +19,125 @@
-        <div class="@hintConfidentialClasses">@T["Password Flow and Client Credentials Flow are allowed."]</div>
-        <div class="@hintConfidentialClasses">@T["Password Flow allows an app to authenticate as an Orchard user providing its username and password Client. This flows allow offline access providing a refresh token."]</div>
-        <div class="@hintConfidentialClasses">@T["Client Credentials allows an app to authenticate without user credentials. Its permissions are set using the roles assigned to the app at the bottom of this screen. Offline access is not allowed."]</div>
-        <div class="@hintConfidentialClasses">@T["Token Requests Url: https:// /Orchard.OpenId/Access/Token"]</div>
-        <div class="@hintConfidentialClasses">@T["Token Request Parameters: grant_type (['password'|'client_credentials']), client_id, client_secret, username, password, resource ('https://yourresourceserver') , scope ('openid, profile, roles, offline_access')"]</div>
-    </div>
-    @{
-        var classes = Model.Type == ClientType.Public ? "form-group" : "form-group collapse";
-    }
-    <div class="@classes" asp-validation-class-for="RedirectUri">
-        <label asp-for="RedirectUri">@T["Redirect Uri"]</label>
-        <span asp-validation-for="RedirectUri" class="text-danger">@T["The Redirect Uri is required."]</span>
-        <input asp-for="RedirectUri" class="form-control" autofocus />
-    </div>
-    <div class="@classes" asp-validation-class-for="LogoutRedirectUri">
-        <label asp-for="LogoutRedirectUri">@T["Logout Redirect Uri"]</label>
-        <span asp-validation-for="LogoutRedirectUri" class="text-danger">@T["The Logout Redirect Uri is required."]</span>
-        <input asp-for="LogoutRedirectUri" class="form-control" autofocus />
-    </div>
-    <div class="@classes" asp-validation-class-for="SkipConsent">
-        <label class="form-check-label">
-            <input asp-for="SkipConsent" type="checkbox" class="form-check-input" checked="@Model.SkipConsent" /> @T["Skip user-consent screen"]
-        </label>
-        <div class="hint">@T["It skips the user-consent screen after login in identity server. When identy server and resource server are the same you usually want to skip this screen."]</div>
+
+    <h3>Flows</h3>
+    <fieldset class="@("form-group" + (openIdSettings.AllowAuthorizationCodeFlow ? "" : " collapse"))" asp-validation-class-for="AllowAuthorizationCodeFlow">
+        <div class="form-check">
+            <label class="form-check-label">
+                <!--There is a known bug with collapse in Bootstrap v4.0.0-alpha.5  https://github.com/twbs/bootstrap/pull/21019 it will be fixed when we upgrade to v4.0.0-alpha.6 -->
+                <a data-toggle="collapse" href="#AllowAuthorizationCodeFlowRecommendedHint" class="a-unstyled">
```

Comment:
This is the reason to add a new rule to TheAdmin.scss .  The only way of using collapse with a checkbox is wrap it with an anchor. The a-unstyled class helps me to avoid to change the style of the text within the checkbox.